pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/MultiStageEngineIntegrationTest.java

```diff
@@ -1721,6 +1721,133 @@ public void testValidateQueryApiError()
     assertFalse(result.get("errorMessage").isNull());
   }
 
+  @Test
+  public void testValidateQueryApiSuccessfulQueries() throws Exception {
+    JsonNode tableConfigsNode = JsonUtils.stringToJsonNode(
+        sendGetRequest(getControllerBaseApiUrl() + "/tables/mytable"));
+    JsonNode schemaNode = JsonUtils.stringToJsonNode(
+        sendGetRequest(getControllerBaseApiUrl() + "/schemas/mytable"));
+
+    String[] successfulQueries = {
+        "SELECT COUNT(*) FROM mytable",
+        "SELECT DivAirportSeqIDs, COUNT(*) FROM mytable GROUP BY DivAirportSeqIDs",
+        "SELECT DivAirportSeqIDs FROM mytable WHERE arrayToMV(DivAirportSeqIDs) > 0 LIMIT 10",
+        "SELECT DivAirportSeqIDs, AirlineID FROM mytable ORDER BY DivAirportSeqIDs LIMIT 5",
+        "SELECT SUM(arrayToMV(DivAirportSeqIDs)) AS total FROM mytable",
+        "SELECT AVG(arrayToMV(DivAirportSeqIDs)) FROM mytable WHERE AirlineID IS NOT NULL"
+    };
+
+    List<String> configs = new ArrayList<>();
+    JsonNode offlineConfig = tableConfigsNode.get("OFFLINE");
+    if (offlineConfig != null && !offlineConfig.isMissingNode() && !offlineConfig.isEmpty()) {
+      configs.add(offlineConfig.toString());
+    }
+    JsonNode realtimeConfig = tableConfigsNode.get("REALTIME");
+    if (realtimeConfig != null && !realtimeConfig.isMissingNode() && !realtimeConfig.isEmpty()) {
+      configs.add(realtimeConfig.toString());
+    }
+    String tableConfigsJson = String.join(",", configs);
+
+    for (String query : successfulQueries) {
+      String requestJson = String.format(
+          "{ \"sql\": \"%s\", \"tableConfigs\": [%s], \"schemas\": [%s], \"ignoreCase\": false }",
+          query, tableConfigsJson, schemaNode.toString());
```

Comment:
We should use `MultiStageQueryValidationRequest` for JSON serde instead of mangling strings (similarly for other tests too).